modules/drivers/snowflake/src/metabase/driver/snowflake.clj

```diff
@@ -459,11 +450,18 @@
 (defmethod sql-jdbc.describe-table/get-table-pks :snowflake
   [_driver ^Connection conn db-name-or-nil table]
   (let [^DatabaseMetaData metadata (.getMetaData conn)]
-    (into [] (sql-jdbc.sync.common/reducible-results
-              #(.getPrimaryKeys metadata db-name-or-nil
-                                (-> table :schema escape-name-for-metadata)
-                                (-> table :name escape-name-for-metadata))
-              (fn [^ResultSet rs] #(.getString rs "COLUMN_NAME"))))))
+    (try
+     (into [] (sql-jdbc.sync.common/reducible-results
+               #(.getPrimaryKeys metadata db-name-or-nil
+                                 (-> table :schema escape-name-for-metadata)
+                                 (-> table :name escape-name-for-metadata))
+               (fn [^ResultSet rs] #(.getString rs "COLUMN_NAME"))))
+     (catch SnowflakeSQLException e
+       ;; dynamic tables doesn't support pks and it'll throw an error if you try to get one
+       (if (= "SQL compilation error: show [constraint] command doesnt support this domain."
+              (ex-message e))
```

Comment:
Relying on error strings staying the same is a brittle assumption. Ideally, we'd know that this table is a dynamic table from the metadata. It seems like you can get the type name from the metadata, based on [the javadocs](https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html#:~:text=getSearchStringEscape()-,getTables,-ResultSet%C2%A0getTables()) of `.getTables`